host/sysfs/onewire.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package sysfs
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"sync"
+)
+
+// OneWire is an open OneWire bus
+type oneWire struct {
+	path         string
+	modProbeCmd  string
+	thermMod     string
+	gpioMod      string
+	masterPrefix string
+	initialized  bool
+}
+
+// OneWireDevice represents a single OneWire device
+type OneWireDevice struct {
+	mtx sync.Mutex
+	f   *os.File
+}
+
+// NewOneWire provides access to OneWire bus on linux devices
+func NewOneWire() (*oneWire, error) {
+	if isLinux {
+		return newOneWire()
+	}
+	return nil, errors.New("sysfs-onewire: not implemented on non-linux OSes")
+}
+
+func newOneWire() (*oneWire, error) {
+	ow := oneWire{
+		path:         "/sys/bus/w1/devices/",
+		modProbeCmd:  "/sbin/modprobe",
+		thermMod:     "w1-therm",
+		gpioMod:      "w1-gpio",
+		masterPrefix: "w1_bus_master",
+		initialized:  false,
+	}
+
+	return &ow, nil
+}
+
+// LoadDrivers makes calls to kernel drivers w1_therm and w1_gpio
```

Comment:
Doc must be much clearer about the side effects, that it shells out. For example when running under systemd as a service the sheer fact of fork()'ing out may be denied.

Calling this function twice will run modprobe twice.

There's no way to initialize the object *without* having modprobe being shelled out.